e2e/sidecars/sidecars_test.go

```diff
@@ -0,0 +1,273 @@
+		var newManifest string
+		It("overwrite existing manifest", func() {
+			stackName := fmt.Sprintf("%s-test-%s", appName, svcName)
+			newManifest = fmt.Sprintf(manifest, sidecarImageURI, stackName)
+			err := ioutil.WriteFile("./copilot/hello/manifest.yml", []byte(newManifest), 0644)
+			Expect(err).NotTo(HaveOccurred(), "overwrite manifest")
+		})
+		It("add addons folder for Firelens permissions", func() {
+			err := os.MkdirAll("./copilot/hello/addons", 0777)
+			Expect(err).NotTo(HaveOccurred(), "create addons dir")
+
+			fds, err := ioutil.ReadDir("./hello/addons")
+			Expect(err).NotTo(HaveOccurred(), "read addons dir")
+
+			for _, fd := range fds {
+				destFile, err := os.Create(fmt.Sprintf("./copilot/hello/addons/%s", fd.Name()))
+				Expect(err).NotTo(HaveOccurred(), "create destination file")
+				defer destFile.Close()
+
+				srcFile, err := os.Open(fmt.Sprintf("./hello/addons/%s", fd.Name()))
+				Expect(err).NotTo(HaveOccurred(), "open source file")
+				defer srcFile.Close()
+
+				_, err = io.Copy(destFile, srcFile)
+				Expect(err).NotTo(HaveOccurred(), "copy file")
+			}
+		})
+	})
+
+	Context("when deploying svc", func() {
+		var (
+			appDeployErr error
+		)
+		BeforeAll(func() {
+			_, appDeployErr = cli.SvcDeploy(&client.SvcDeployInput{
+				Name:     svcName,
+				EnvName:  "test",
+				ImageTag: "gallopinggurdey",
+			})
+		})
+
+		It("svc deploy should succeed", func() {
+			Expect(appDeployErr).NotTo(HaveOccurred())
+		})
+
+		It("svc show should include a valid URL and description for test env", func() {
+			svc, svcShowErr := cli.SvcShow(&client.SvcShowRequest{
+				AppName: appName,
+				Name:    svcName,
+			})
+			Expect(svcShowErr).NotTo(HaveOccurred())
+			Expect(len(svc.Routes)).To(Equal(1))
+
+			// Call each environment's endpoint and ensure it returns a 200
+			route := svc.Routes[0]
+			Expect(route.Environment).To(Equal("test"))
+			uri := route.URL + "/health-check"
+
+			// Service should be ready.
+			var resp *http.Response
+			var fetchErr error
+			Eventually(func() (int, error) {
+				resp, fetchErr = http.Get(uri)
+				return resp.StatusCode, fetchErr
+			}, "30s", "1s").Should(Equal(200))
+
+			// Read the response - our deployed apps should return a body with their
+			// name as the value.
+			bodyBytes, err := ioutil.ReadAll(resp.Body)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(bodyBytes)).To(Equal("Ready"))
+		})
+
+		It("svc logs should display logs", func() {
+			var svcLogs []client.SvcLogsOutput
+			var svcLogsErr error
+			Eventually(func() ([]client.SvcLogsOutput, error) {
+				svcLogs, svcLogsErr = cli.SvcLogs(&client.SvcLogsRequest{
+					AppName: appName,
+					Name:    svcName,
+					EnvName: "test",
+					Since:   "1h",
+				})
+				return svcLogs, svcLogsErr
+			}, "60s", "10s").ShouldNot(BeEmpty())
+
+			for _, logLine := range svcLogs {
+				Expect(logLine.Message).NotTo(Equal(""))
+				Expect(logLine.LogStreamName).NotTo(Equal(""))
```

Comment:
maybe we can validate that the logstreamname is the custom one that firelens set for us?

Reply:
But the thing is it contains not only the logs from firelens but also nginx logs and the logs for firelens itself...

Reply:
maybe i can try to check if the log stream name falls into these three options